### DISCUSSION TITLE
Inaccurate fluxes values of MSRE

### DISCUSSION THREAD

**Discussion by saqibhayat556** [arfc/moltres#241] [Q&A Reactor multiphysics] [answered]
@smpark7  hello sir, Whenever i run the given example of MSRE in directory (projects/moltres/tests/twod_axi_coupled) and visualize its output in Paraview. i don't get the desired values of fluxes although profile of fluxes is fine but the value of fluxes is not accurate. Same for precursor concentration. My flux should be in 10^13 range but i get the value of 6.5. I have tried running example as it is without any changes. Below screenshot of group1 flux is attached.
![flux](https://github.com/arfc/moltres/assets/124503670/00e61a8a-5e8c-45fe-9511-c7fbe579058e)

**Comment by smpark7** [comment]
Hey @saqibhayat556, this is intended behavior from Moltres. At the top of the input file, there is a `nt_scale` parameter that goes into the `TransientFissionHeatSource` kernel further down the input file. This parameter scales the neutron fluxes down by 1e13 so that the fluxes and temperatures are closer in magnitude. This was an early feature to help the solver converge faster. Nowadays, I prefer using the automatic scaling feature (from MOOSE) which automatically scales the variable values used in the solver while keeping the actual variable values as is.

**Reply by saqibhayat556** [reply]
Thanks @smpark7 for taking out time to address my query! 
The power of reactor is not mentioned anywhere in input file. How do we know the value of nt_scale that is being used to scale down the fluxes by 1e13 furthermore, what changes should we do in our input file to visualize true value of fluxes at output and how to do automatic scaling?

**Reply by smpark7** [reply]
This example is a transient simulation which allows the power to vary over time towards a steady state value. The model parameters have been tweaked to match the approximate power output of the actual MSRE.

To visualize the true flux values, you can 1) change the 4th line from `nt_scale=1e13` to `nt_scale=1` or 2) delete the 4th and 145th lines which define `nt_scale` values (Moltres defaults to `nt_scale=1` if it is not provided). Refer to my instructions [here](https://arfc.github.io/software/moltres/wiki/recommended_settings/) for enabling automatic scaling.

Note: We're currently in the process of transitioning the Moltres documentation website from https://arfc.github.io/software/moltres/wiki/ to https://arfc.github.io/moltres/ . Once complete, you may find the instructions on the newer weblink.

**Reply by saqibhayat556** [reply]
@smpark7 Thanks for the help!
 Tried this but it did not work. My multiplication factor starts increasing from time step of 8 seconds and increases upto 5 at time step of around 18 seconds and then it turns negative due to negative current.  below is the input file attached

**Reply by smpark7** [reply, marked as answer]
Hey @saqibhayat556
I tested it myself and found that my instructions didn't work. The issue stems from:
1. taking a timestep size that is larger than the time it takes for the salt to travel from the bottom to the top of the model (~6.9s)
2. not scaling the initial condition up by 1e13.

The solver can't converge well under those conditions. Limiting the maximum timestep size to 5 seconds fixes this. Here are the changes I made to the original input file:

```
diff --git a/tests/twod_axi_coupled/auto_diff_rho.i b/tests/twod_axi_coupled/auto_diff_rho.i
index 319bf613..7f366a77 100644
--- a/tests/twod_axi_coupled/auto_diff_rho.i
+++ b/tests/twod_axi_coupled/auto_diff_rho.i
@@ -1,7 +1,7 @@
 flow_velocity=21.7      # cm/s. See MSRE-properties.ods
 ini_temp=922
 diri_temp=922
-nt_scale=1e13
+nt_scale=1

 [GlobalParams]
   num_groups = 2
@@ -18,18 +18,18 @@ nt_scale=1e13
   [./group1]
     order = FIRST
     family = LAGRANGE
-    initial_condition = 1
-    scaling = 1e4
+    initial_condition = 1e13
   [../]
   [./group2]
     order = FIRST
     family = LAGRANGE
-    initial_condition = 1
-    scaling = 1e4
+    initial_condition = 1e13
   [../]
   [./temp]
     initial_condition = ${ini_temp}
-    scaling = 1e-4
   [../]
 []

@@ -241,8 +241,13 @@ nt_scale=1e13
   steady_state_detection = true
   steady_state_tolerance = 1e-10

+  automatic_scaling = true
+  resid_vs_jac_scaling_param = 0.2
+  scaling_group_variables = 'group1 group2; pre1 pre2 pre3 pre4 pre5 pre6; temp'
+  compute_scaling_once = false
+
   dtmin = 1e-5
-  # dtmax = 1
+  dtmax = 5
   # dt = 1e-3
   [./TimeStepper]
     type = IterationAdaptiveDT
```
Let me know if you still face issues.

**Reply by saqibhayat556** [reply]
Thank you so much @smpark7 for your help! It worked.
My question is, what model parameters i have to tweak to run my reactor at higher power level (let say 15MW) and to visualize corresponding flux for this power level?

**Reply by smpark7** [reply]
I'm glad to help! You can increase the flow rate in line 1 or decrease the inlet temperature in line 3. Note that significant changes to either parameter may affect solver convergence.

**Reply by saqibhayat556** [reply]
Thank you!
